### DISCUSSION TITLE
TCP encapsulation according to RFC 8229

### DISCUSSION THREAD

**Discussion by platz** [strongswan/strongswan#220] [Q&A] [answered]
Hello Strongswan Team,

RFC 8229 is now part of Linux kernel 5.6, as far as i understood.
Are there any plans to implement access to this new kernel functionality from the user land code, the IKE daemon? That it would be possible to configure something like "espintcp=yes" in ipsec.conf or swanctl.conf.

If one consider to do an own implementation of it, where in the Strongswan code is a good place to start and on what should be paid attention on?

Thanks in advance,
Alex

**Comment by strongX509** [comment, marked as answer]
Hello Alex,

support of RFC 8229 has a high priority on our roadmap and will be implemented this year.

Best regards

Andreas

**Reply by TomHsiung** [reply]
Hi, Strong,

How's RFC 8229 project going?
